internal/pkg/application/getdevice.go

```diff
@@ -38,8 +38,8 @@ func GetDeviceStatusAndSendReportIfMissing(log logging.Logger, baseUrl string, i
 				storedValue := deviceStatusCache[device.ID]
 
 				if device.Value.Value == "off" && device.Value.Value != storedValue {
-
-					log.Infof("state changed to \"off\" for device %s", device.ID)
+					sublogger := log.With().Str("deviceID", device.ID).Logger()
```

Comment:
Sub loggers are usually used in a larger scope than "next line". Do we need it for the whole function maybe?